### PR TITLE
Refactored `asy` renderer and added tests

### PR DESCRIPTION
**Pull request by jmmshn**
# Refactored `asy` renderer
- More sensible parsing of user and default settings for scene attributes, also standardized how things were read.
- Allow users a way to update defaults for a scene by modifying the `Scene._meta["asy"]`
- Added tests

### CODE REVIEW

**Comment by jmmshn** [comment]
OK, so module tests are passing now and it's failing the same web tests as before.  @janosh , @mkhorton should be ok to merge now.  Also I just made this with it pretty easily thought this looked cool :)

image

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ docs_rst/_build
 build/*
 *.egg-info
 dist/*
+.vscode
```
Comment (janosh) [inline .gitignore:15]: this should probably go in your global `gitignore`

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+from crystal_toolkit.core.scene import Cylinders, Lines, Spheres, Surface
+
+
+@pytest.fixture(scope="session")
+def test_files():
+    return Path(__file__).parent / "test_files"
+
+
+@pytest.fixture(scope="session")
```
Comment (janosh) [inline tests/conftest.py:16]: If only used in `tests/asy_test/test_asymptote.py`, maybe we should move the fixture there?

File: .gitignore (hunk 1 above)
Comment (jmmshn) [reply .gitignore:15]: IMO if our code doe not do anything with VS code then it should go in here, right?  That's one less step for everyone that's working on the code.

File: tests/conftest.py (hunk 2 above)
Comment (jmmshn) [reply tests/conftest.py:16]: Makes sense I will move it.

**Comment by janosh** [comment]
Looks like a lot of great work! Thanks for the clean up. Question before merging: Is the `tests/test_chgcar.py` deletion intentional?

File: .gitignore (hunk 1 above)
Comment (janosh) [reply .gitignore:15]: I use VS Code too. I put that line in my global `gitignore` so I don't have to add it to every project. Overall less work I think but definitely fine to leave it in. :)

**Comment by jmmshn** [comment]
> Looks like a lot of great work! Thanks for the clean up. Question before merging: Is the `tests/test_chgcar.py` deletion intentional?

Yes, that was not really a test more of an app so it was actually testing anything

**Pull request merged**